segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -46,38 +47,34 @@ public IndexAppendProcessor(ScheduledExecutorService indexSegmentUpdateExecutor,
         this.store = store;
     }
 
-    /**
-     * Appends index segment on a separate thread.
-     * @param segmentName  segment name.
-     */
-    protected void processAppend(String segmentName) {
+    protected void processAppend(IndexAppend indexAppend) {
         // Not updating index segment for transient and transaction type.
-        if (isTransientSegment(segmentName) || isTransactionSegment(segmentName)) {
+        if (isTransientSegment(indexAppend.getSegment()) || isTransactionSegment(indexAppend.getSegment())) {
             return;
         }
-        LatestItemSequentialProcessor<String> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
-        latestItemProcessor.updateItem(segmentName);
+        LatestItemSequentialProcessor<IndexAppend> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
+        latestItemProcessor.updateItem(indexAppend);
     }
 
-    private void handleIndexAppend(String segmentName) {
+    private void handleIndexAppend(IndexAppend indexAppend) {
         //TODO : Update the index segment Attribute as needed going forward. If eventcount for main segment is null then we are passing it as 0 in index segment.
         AttributeUpdateCollection attributes = AttributeUpdateCollection.from(
+                //TODO: ReplaceIfGreater
```

Comment:
@tkaitchuck we are already having a check in place where we are checking if the main segment append is success then only we are proceeding to append to the index segment. Do we still need to add the AttributeUpdate parameter here ? If yes, can you please tell what condition can we use here ?

Reply:
There may be multiple appends in parallel from different writers. We need to use the attributes to make sure that the appends to the index segment go in order or not at all.